qiskit/optimization/converters/integer_to_binary.py

```diff
@@ -54,15 +55,14 @@ def __init__(self) -> None:
         self._src = None  # type: Optional[QuadraticProgram]
         self._dst = None  # type: Optional[QuadraticProgram]
         self._conv = {}  # type: Dict[Variable, List[Tuple[str, int]]]
+
         # e.g., self._conv = {x: [('x@1', 1), ('x@2', 2)]}
 
-    def encode(self, op: QuadraticProgram, name: Optional[str] = None) -> QuadraticProgram:
```

Comment:
It might be good to deprecate these methods instead of just removing them since the optimization module is already released since 3 months and might be used by several people. Could you add deprecation warnings saying that the new method is now `convert`? See e.g. here for how deprecation warnings usually look like: https://github.com/Qiskit/qiskit-aqua/blob/c343e52597e11a08bb20eadfff11bf93fab974e8/qiskit/aqua/components/variational_forms/ry.py#L94

You could also move that directly inside the `QuadraticProgramConverter` base class (same for `decode`/`interpret`) so that you don't have to write it for every converter.